packages/in_app_purchase/in_app_purchase/example/lib/readme_excerpts.dart

```diff
@@ -0,0 +1,64 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// ignore_for_file: avoid_print
+
+import 'package:in_app_purchase/in_app_purchase.dart';
+// #docregion AndroidProduct
+import 'package:in_app_purchase_android/billing_client_wrappers.dart';
+import 'package:in_app_purchase_android/in_app_purchase_android.dart';
+// #enddocregion AndroidProduct
+
+// #docregion IOSProduct
+// #docregion RedeemOffer
+import 'package:in_app_purchase_storekit/in_app_purchase_storekit.dart';
+// #enddocregion RedeemOffer
+import 'package:in_app_purchase_storekit/store_kit_wrappers.dart';
+// #enddocregion IOSProduct
+
+/// Demonstrates getting platform-specific attributes from product
+/// and purchase details, for the README.
+void platformSpecific() {
+  const List<ProductDetails> productDetails = <ProductDetails>[];
+  const List<PurchaseDetails> purchaseDetails = <PurchaseDetails>[];
```

Comment:
It seems very strange to declare these as lists and then check them for an unrelated type just below. I know it's not going to show up in the README, but having this code be internally incoherent could make it hard to maintain the excerpts. Can't we declare these as the relevant objects rather than as lists of those objects?